src/coreComponents/dataRepository/DataContext.hpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file DataContext.hpp
+ */
+
+#ifndef GEOS_DATAREPOSITORY_SOURCECONTEXT_HPP_
+#define GEOS_DATAREPOSITORY_SOURCECONTEXT_HPP_
+
+#include "common/DataTypes.hpp"
+#include "common/Logger.hpp"
+#include "xmlWrapper.hpp"
+
+namespace geos
+{
+
+namespace dataRepository
+{
+
+class Group;
+class WrapperBase;
+
+/// This abstract class stores data that helps to retrieve an object:
+/// - from which position in a file (if applicable), see DataFileContext,
+/// - where it is located in the data hierarchy, see GroupContext and WrapperContext.
+/// Typically, the target object contain an unique_ptr< DataContext > instance of this class.
+class DataContext
+{
+public:
+
+  /**
+   * @brief Construct a new DataContext object.
+   * @param objectName the target object name
+   * @param isDataFileContext true if this Context is a DataFileContext (see isDataFileContext for more infos)
+   */
+  DataContext( string const & objectName, bool isDataFileContext );
+
+  /**
+   * @return A string that mention all the known informations to retrieve from where the target
+   * object comes from.
+   */
+  virtual string toString() const = 0;
+
+  string getObjectName() const
+  { return m_objectName; }
+
+  /**
+   * @brief In some cases, we need to know if a DataContext is from a file. It means that it
+   * is a more user-friendly information compared to other DataContext classes.
+   * @return true if the context is from a file.
+   */
```

Comment:
/**
   * @brief Flag on availability of a file. Used to provide more user-friendly information 
   * @return true if the context is built from a file.
   */